lib/tasks/TaskScheduler.js

```diff
@@ -84,8 +103,57 @@ class TaskScheduler {
                 return queue.push(ctx, onTaskEnd);
             }
         }
+        if (this._noQueueKeyTasks) {
+            return this._noQueueKeyTasks.push(ctx, onTaskEnd);
+        }
         return process.nextTick(() => this._processingFunc(ctx, onTaskEnd));
     }
+
+    /**
+     * @returns {Number} The number of tasks waiting to be processed
+     */
+    length() {
+        const noQueueKeyTasksLength = this._noQueueKeyTasks ? this._noQueueKeyTasks.length() : 0;
+        const taskQueuesLength = Object.keys(this._taskQueues)
+                .reduce((acc, key) => acc + this._taskQueues[key].length(), 0);
```

Comment:
`length()` will be called many times, not sure it is worth doing the computation every time: better to just update the `length` whenever there is a push or a completion